test/python/channelwidth.py

```diff
@@ -0,0 +1,405 @@
+
+#---------------------------------------------------------
+# Channel width in y-direction
+#---------------------------------------------------------
+wc_test.Solver.Nonlinear.ChannelWidthExistY = True
+wc_test.ChannelWidthY.Type            = 'Constant'
+wc_test.ChannelWidthY.GeomNames       = 'domain'
+wc_test.ChannelWidthY.Geom.domain.Value = 1.0
+
+#-----------------------------------------------------------------------------
+# Run ParFlow
+#-----------------------------------------------------------------------------
+base = os.path.join(os.getcwd(), "output/constant")
+mkdir(base)
+print(f"base: {base}")
+wc_test.run(working_directory=base)
+
+#-----------------------------------------------------------------------------
+# Check if values are as expected
+#-----------------------------------------------------------------------------
+
+data_wcx = read_pfb(os.path.join(base, "channel_width_example.out.wc_x.pfb"))
+data_wcy = read_pfb(os.path.join(base, "channel_width_example.out.wc_y.pfb"))
+
+ones_array = np.ones((1, 1, 20))
+
+if not np.array_equal(data_wcx, ones_array) or not np.array_equal(data_wcy, ones_array): 
+    print(f"{run_name} : FAILED")
+    sys.exit(1)
+
+#---------------------------------------------------------
+# Test channel width with PFB input
+#---------------------------------------------------------
+
+base = os.path.join(os.getcwd(), "output/pfb")
+mkdir(base)
+print(f"base: {base}")
+write_pfb(get_absolute_path(base + '/Channel_Width_X.pfb'), ones_array)
+write_pfb(get_absolute_path(base + '/Channel_Width_Y.pfb'), ones_array)
+
+#---------------------------------------------------------
+# Channel width in x-direction
+#---------------------------------------------------------
+wc_test.Solver.Nonlinear.ChannelWidthExistX = True
+wc_test.ChannelWidthX.Type            = 'PFBFile'
+wc_test.ChannelWidthX.GeomNames       = 'domain'
+wc_test.ChannelWidthX.FileName        = 'Channel_Width_X.pfb'
+
+#---------------------------------------------------------
+# Channel width in y-direction
+#---------------------------------------------------------
+wc_test.Solver.Nonlinear.ChannelWidthExistY = True
+wc_test.ChannelWidthY.Type            = 'PFBFile'
+wc_test.ChannelWidthY.GeomNames       = 'domain'
+wc_test.ChannelWidthY.FileName        = 'Channel_Width_Y.pfb'
+
+#-----------------------------------------------------------------------------
+# Run ParFlow
+#-----------------------------------------------------------------------------
+
+wc_test.run(working_directory=base)
+
+#-----------------------------------------------------------------------------
+# Check if values are as expected
+#-----------------------------------------------------------------------------
+
+data_wcx = read_pfb(os.path.join(base, "channel_width_example.out.wc_x.pfb"))
+data_wcy = read_pfb(os.path.join(base, "channel_width_example.out.wc_y.pfb"))
+
+if not np.array_equal(data_wcx, ones_array) or not np.array_equal(data_wcy, ones_array):
```

Comment:
This test in not checking very much.  Should check simulation results.

Reply:
This PR simply adds the capability to read ChannelWidthX and ChannelWidthY inputs. It is not yet integrated with overland flow. @reedmaxwell suggested breaking this into two separate PRs, because the new overland flow formulation will need different boundary conditions as well and this might take longer to develop. So this test just checks that ParFlow reads correctly the inputs. It can be modified once the integration with overland flow happens, to check against simulation results.